src/core/util/lazy_docs.pl

```diff
@@ -0,0 +1,27 @@
+:- module('util/lazy_docs', [stream_to_lazy_docs/2]).
+
+:- use_module(library(http/json)).
+:- use_module(library(lists)).
+
+stream_to_lazy_docs(Stream, List) :-
+    put_attr(List, 'util/lazy_docs', lazy_input(Stream, _)).
+
+attr_unify_hook(State, Value) :-
+    State = lazy_input(Stream, Peak),
+    (   var(Peak)
+    ->  json_read_dict(Stream, Term, [default_tag(json), end_of_file(eof)]),
+        (   Term = eof
+        ->  nb_setarg(2, State, []),
+            Value = []
+        ;   is_list(Term)
+        ->  stream_to_lazy_docs(Stream, Rest),
+            append(Term,Rest,Result),
+            nb_linkarg(2, State, Result),
+            Value = Result
+        ;   stream_to_lazy_docs(Stream, Rest),
+            Result = [Term|Rest],
+            nb_linkarg(2, State, Result),
+            Value = Result
+        )
```

Comment:
one thing i'm a bit confused about.
It is my understanding that attributed variables disappear the moment you unify the variable. So wouldn't your state term basically disappear cause you're also unifying Value in all these cases?
Or is this a case where you're setting things up so that you save the value in case a rollback happens?

Reply:
Just for rollback on failure of unification.